File: Stacks and Queues/2. Queue Implementation/QueueUsingArray.java
import java.io.*;

public class QueueUsingArray
{
    private int arr[];
    private int front, end;

    public QueueUsingArray(int maxsize)
    {
        this.front = 0;
        this.end = 0;
        arr = new int[maxsize];
    }

    public boolean isEmpty()
    {
        return (front == end);
    }

    public void enqueue(int val)
    {
        int len = arr.length;
        if (len == end)
        {
            System.out.println("Queue is FULL.");
            return;
        }
        arr[end++] = val;
    }

    public int dequeue()
    {
        if (isEmpty())
        {
            throw new RuntimeException("Queue is Empty.");
        }
        return (arr[front++]);
    }

    public int peek()
    {
        if (isEmpty())
            throw new RuntimeException("Queue is Empty.");
        return (arr[front]);
    }

    public static void main(String args[])
    {
        QueueUsingArray q = new QueueUsingArray(10);
        for (int i = 0; i < 10; i++)
        {
            q.enqueue(i+1);
        }

        while(!q.isEmpty())
            System.out.print(q.dequeue() + " ");
        
        System.out.println();
    }
}

File: Stacks and Queues/2. Queue Implementation/QueueUsingArray.py
__author__ = 'Kalpak Seal'

class Queue:
    def __init__(self):
        self.queue = []

    def isEmpty(self):
        return self.queue == []

    def enqueue(self,data):
        self.queue.append(data)

    def dequeue(self):
        return self.queue.pop(0)

    def size(self):
        return len(self.queue)

    def print(self):
        for i in self.queue:
            print (i, end=' ')
        print()

if __name__ == '__main__':
    q = Queue()
    q.enqueue(10)
    q.enqueue(20)
    q.enqueue(30)

    print("After enqueueing 3 elements: ")
    q.print()

    print("Dequeueing an element: ", q.dequeue())

    print("Presently the queue is: ")
    q.print()